blank_react_project/src/index.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+import getConfig from './config.js';
+import * as nearlib from 'nearlib';
+
+async function doInitContract() {
+    window.nearConfig = getConfig('development')
+    console.log("nearConfig", window.nearConfig);
+
+    // Initializing connection to the NEAR node.
+    window.near = await nearlib.connect(Object.assign(window.nearConfig, { deps: { keyStore: new nearlib.keyStores.BrowserLocalStorageKeyStore() } }));
+    // Needed to access wallet login
+    window.walletAccount = new nearlib.WalletAccount(window.near);
+
+
+    // Getting the Account ID. If unauthorized yet, it's just empty string.
+    window.accountId = window.walletAccount.getAccountId();
+
+    // Initializing near and near client from the nearlib.
+    // window.near = new nearlib.Near(new nearlib.NearClient(
+    //   window.walletAccount,
+    //   // We need to provide a connection to the blockchain node which we're going to use
+    //   new nearlib.LocalNodeConnection(nearConfig.nodeUrl),
+    // ));
+
+    window.contract = await window.near.loadContract(window.nearConfig.contractName, {
+        viewMethods: [
+            "hello",],
+        changeMethods: [
+            ],
+        sender: window.accountId
+    });
+}
+
+window.nearInitPromise = doInitContract().then(() => {
+    ReactDOM.render(<App contract={window.contract} wallet={window.walletAccount}/>,
```

Comment:
app rendering shouldn't be delayed to until contract is setup as then user won't see app UI until it happens, which is undesirable

Reply:
but it has to be contract ready and then to render dom, so I think it is okey since corgi also present like this and no blank UI happen